lib/Optimizer/Dialect/Quake/QuakeTypes.cpp

```diff
@@ -88,6 +88,39 @@ void quake::StruqType::print(AsmPrinter &printer) const {
   printer << '>';
 }
 
+// This recursive function returns true if and only if \p ty is a quake
+// reference type and the number of qubits is a compile-time known constant.
```

Comment:
A quake reference type is a type in the set `{ ref, veq, struq }`. Any other type is not a reference type.

This checks that the composition of types does not contain a `veq` of unspecified size.

Reply:
OK, that makes sense in that context. It just threw me for a loop that it was meaning "quake reference type" in the categorical sense ... not specifically referring to `quake::RefType`. Since they are so similar, I was thinking they were talking about the same thing.

Reply:
Fixed in 5ca16fe5e309359a337123c678e1866112473e02